src/site/content/en/learn/html/template/index.md

```diff
@@ -3,10 +3,312 @@ title: 'Template, slot, and shadow'
+shadow root containing all the contents of the template we associate with it. The `<slot>` elements from the template are replaced
+with the contents of the element within the `<star-rating>` whose slot attribute value matches the `<slot>`'s name value, if
+there is one. If not, the contents of the template's slots are displayed.
+
+Content within a custom element that isn't associated with a slot—the `<p>` in our third `<star-rating>`—is not included in
+the shadow root and therefore not displayed.
+
+We [define the custom element](https://developer.mozilla.org/docs/Web/Web_Components/Using_custom_elements) named `star-rating`
+by extending the `HTMLElement`:
+
+```javascript
+customElements.define('star-rating',
+    class extends HTMLElement {
+        constructor() {
+            super(); // Always call super first in constructor
+            const starRating = document.getElementById('star-rating-template').content;
+            const shadowRoot = this.attachShadow({
+                mode: 'open'
+            });
+            shadowRoot.appendChild(starRating.cloneNode(true));
+        }
+    });
+```
+
+{% Codepen {
+user: 'web-dot-dev',
+id: 'poOzjpj'
+} %}
+
+Now that the element is defined, every time the browser encounters a `<star-rating>` element, it will be rendered as defined
+by the element with the `#star-rating-template`, which is our template. The browser will attach a shadow DOM tree to the node, appending
+a [clone](https://developer.mozilla.org/docs/Web/API/Node/cloneNode) of the template contents to that shadow DOM.
+Note that the elements upon which you can [`attachShadow()` are limited](https://developer.mozilla.org/docs/Web/API/Element/attachShadow#elements_you_can_attach_a_shadow_to).
+
+```javascript
+const shadowRoot = this.attachShadow({mode: 'open'});
+shadowRoot.appendChild(starRating.cloneNode(true));
+```
+
+If you take a look at the developer tools, you'll note the `<form>` from the `<template>` is part of the shadow root of each custom element.
+A clone of the `<template>` contents is apparent in each custom element in the developer tools and visible in the browser, but the contents
+of the custom element itself are not rendered to the screen.
+
+{% Img src="image/cGQxYFGJrUUaUZyWhyt9yo5gHhs1/CqbHMMxnse6n8STAocFo.png", alt="DevTools screenshot showing the cloned template contents in each custom element.", width="800", height="842" %}
+
+In the `<template>` example, we appended the template contents to the document body, adding the content to the regular DOM.
+In the [`customElements` definition](https://developer.mozilla.org/docs/Web/API/CustomElementRegistry/define), we used the same
+[`appendChild()`](https://developer.mozilla.org/docs/Web/API/Node/appendChild), but the cloned template contents were appended to an
+encapsulated shadow DOM.
+
+Being part of a shadow DOM rather than the standard DOM, the styling within Codepen's CSS tab does not apply. The CSS tab
```

Comment:
```suggestion
Notice how the stars went back to being unstyled radio buttons? Being part of a shadow DOM rather than the standard DOM, the styling within Codepen's CSS tab does not apply. That tab's CSS
```